src/hypertable_insert.c

```diff
@@ -36,35 +59,35 @@ static void
 hypertable_insert_begin(CustomScanState *node, EState *estate, int eflags)
 {
 	HypertableInsertState *state = (HypertableInsertState *) node;
+	ModifyTableState *mtstate;
 	PlanState *ps;
+	bool found_cdstate = false;
+	int i;
 
 	ps = ExecInitNode(&state->mt->plan, estate, eflags);
 	node->custom_ps = list_make1(ps);
 
-	if (IsA(ps, ModifyTableState))
-	{
-		ModifyTableState *mtstate = (ModifyTableState *) ps;
-		int i;
-
-		/*
-		 * Find all ChunkDispatchState subnodes and set their parent
-		 * ModifyTableState node
-		 */
-		for (i = 0; i < mtstate->mt_nplans; i++)
-		{
-			if (IsA(mtstate->mt_plans[i], CustomScanState))
-			{
-				CustomScanState *csstate = (CustomScanState *) mtstate->mt_plans[i];
+	Assert(IsA(ps, ModifyTableState));
+	mtstate = (ModifyTableState *) ps;
```

Comment:
This does assertion and casting in one go:
```suggestion
	mtstate = castNode(ModifyTableState, ps);
```